lib/signaling/v2/peerconnection.js

```diff
@@ -634,6 +634,25 @@ function getConfiguration(configuration) {
 }
 
 /**
+ * Get the m= sections of a particular kind and direction from an sdp.
+ * @param {string} sdp -  sdp string
+ * @param {string} [kind] - Pattern for matching kind
+ * @param {string} [direction] - Pattern for matching direction
+ * @returns {Array<string>} mediaSections
+ */
+function getMediaSections(sdp, kind, direction) {
+  kind = kind || '.*';
+  direction = direction || '.*';
+  return sdp.split('\r\nm=').slice(1).map(function(mediaSection) {
+    return 'm=' + mediaSection;
+  }).filter(function(mediaSection) {
+    var kindPattern = new RegExp('m=' + kind, 'gm');
+    var directionPattern = new RegExp('a=' + direction, 'gm');
```

Comment:
@ryan-rowland I agree. I'll make a note to refactor this. But for now I don't want to introduce code changes to this function because I would also have to update the corresponding method in the webrtc shim project and submit a PR for that as well. I will do this as part of a future PR.